packages/client-node/src/config.ts

```diff
@@ -22,10 +24,21 @@ export type NodeClickHouseClientConfigOptions =
       enabled?: boolean
       /** For how long keep a particular idle socket alive on the client side (in milliseconds).
        *  It is supposed to be a fair bit less that the ClickHouse server KeepAlive timeout,
-       *  which is by default 3000 ms for pre-23.11 versions.
+       *  which is by default 3000 ms for pre-23.11 versions. <br/>
+       *  When set to `0`, the idle socket management feature is disabled.
        *  @default 2500 */
       idle_socket_ttl?: number
     }
+    /** Custom HTTP agent to use for the outgoing HTTP(s) requests.
+     *  If set, {@link BaseClickHouseClientConfigOptions.max_open_connections}, {@link tls} and {@link keep_alive}
+     *  options have no effect, as it is part of the default underlying agent configuration.
+     *  @experimental - unstable API, might be a subject to change in the future; please provide your feedback in the repository.
+     *  @default undefined */
+    http_agent?: http.Agent | https.Agent
+    /** Enable or disable the `Authorization` header with basic auth for the outgoing HTTP(s) requests.
+     *  @experimental - unstable API, might be a subject to change in the future; please provide your feedback in the repository.
+     *  @default true (enabled) */
+    set_basic_auth_header?: boolean
```

Comment:
what if we change the client signature
```typescript
  /** The name of the user on whose behalf requests are made.
   *  @default default */
  username?: string | null
  /** The user password.
   *  @default empty string */
  password?: string | null
```

if `null` is provided, no defaults are applied, and no auth header is set?